### ISSUE TITLE
Error: Dynamic server usage: headers for statically generated pages 

### ISSUE THREAD

**Issue by nmitic** [nextauthjs/next-auth#8359]
### Question 💬

First time trying NextAuth and I can not get it to work on SSG.

This is the error I am getting when trying to access one of the SSG pages.

```
Dynamic server usage: headers
```

I understand that `getServerSession` can not be used on the SSG, so I am sure I am getting something wrong here. Ideally I would not need to convert components into the client one in order to show some user related data.

I have looked in many issues similar to this one, but none seems to explain how to do NextAuth with SSG content.

### How to reproduce ☕️

Here is repo that produces the error https://github.com/nmitic/nikola.mitic.dev/blob/auth/app/layout.tsx#L18

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by zephimir** [comment]
@nmitic  did you find a solution for this?

**Comment by balazsorban44** [comment]
For static pages, you can use middleware in front. Auth depends on the request, so a static page cannot be authenticated from within the page component.

**Comment by rharkor** [comment]
Hi,
@zephimir @balazsorban44 

I encounter a similar problem where I was checking the user session in my `page.tsx` and ending up with `DynamicServerError - Dynamic Server Usage` at build time. In order to fix this you need to tell next js that you will use the headers in this page.

TLDR;
```tsx
export default async function Profile() {
    // Add this before the breaking call
    headers()
    // The call that require authentication
    const account = await serverTrpc.me.getAccount()
}
```
